hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/MountTableResolver.java

```diff
@@ -381,18 +417,37 @@ public void clear() {
   public PathLocation getDestinationForPath(final String path)
       throws IOException {
     verifyMountTable();
+    PathLocation res;
+    String tmpPath = path;
```

Comment:
This whole pass to finalPath could be a method and the names could be a little more self-explanatory.